core/imageroot/var/lib/nethserver/node/actions/update-core/95cleanup_images

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+#
+# Copyright (C) 2022 Nethesis S.r.l.
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+exec 1>&2
+set -e
+
+# Clean up local images that are no longer required by the new core
+# version
+
+export LC_ALL=C # required by sort for portability
+
+source /etc/nethserver/core.env
+
+# Find the images that are no longer required:
+# 1. generate a sorted list starting with the old module image ID,
+#    followed by its additional images, as declared by the
+#    org.nethserver.images label
+# 2. similarly, generate the list for the new image ID
+# 3. compare (diff) the lists and extract the images that are no longer
+#    required
+# 4. pass them to `podman rmi`
+diff  <(
+    echo "${PREV_IMAGE_ID}"
+    podman inspect -f json "${PREV_IMAGE_ID:-none}" | \
+            jq -r '.[0].Config.Labels."org.nethserver.images" | split(" ") | join("\n")' | \
+            sort
+) <(
+    echo "${IMAGE_ID}"
+    podman inspect -f json "${IMAGE_ID:-none}" | \
+            jq -r '.[0].Config.Labels."org.nethserver.images" | split(" ") | join("\n")' | \
+            sort
+) | sed -n '/< / s/< //p' | \
```

Comment:
I'm not really happy with the `sed` part, maybe it's better to use a safer diff output?

Still, I'm not sure it worth the change.
```suggestion
diff -u <(
    echo "${PREV_IMAGE_ID}"
    podman inspect -f json "${PREV_IMAGE_ID:-none}" | \
            jq -r '.[0].Config.Labels."org.nethserver.images" | split(" ") | join("\n")' | \
            sort
) <(
    echo "${IMAGE_ID}"
    podman inspect -f json "${IMAGE_ID:-none}" | \
            jq -r '.[0].Config.Labels."org.nethserver.images" | split(" ") | join("\n")' | \
            sort
) | grep -e '^-' | cut -d '-' -f2
```

Reply:
Ok to add the regexp anchor. About `-u` I prefer to use the normal diff format, like similar code here:
https://github.com/NethServer/ns8-core/blob/8efbdeb19f6de6145531142b4d5ac90a36a24fc7/core/imageroot/usr/local/agent/bin/install-coreimage#L49